### ISSUE TITLE
Overriding the Starlark aar_import label doesn't seem to work

### ISSUE THREAD

**Issue by ted-xie** [bazelbuild/rules_jvm_external#1131]
In rules_android, I'm trying to get rules_jvm_external to use the local rules_android repo's version of aar_import().

You can see my attempt [here](https://github.com/ted-xie/rules_android/commit/f3dc5dfa70bc99cb066964f2d354f7e332e0a2d6). With bzlmod disabled (see changes to defs.bzl), I can get `USE_BAZEL_VERSION=last_green bazelisk build @rules_android_maven//... --noenable_bzlmod` to pass. However, with bzlmod enabled, I get this error:

```
ERROR: Skipping '@rules_android_maven//...': error loading package under directory '': error loading package '@@rules_jvm_external~~maven~rules_android_maven//': Unable to find package for @@[unknown repo 'rules_android' requested from @@rules_jvm_external~~maven~rules_android_maven]//android:rules.bzl: The repository '@@[unknown repo 'rules_android' requested from @@rules_jvm_external~~maven~rules_android_maven]' could not be resolved: No repository visible as '@rules_android' from repository '@@rules_jvm_external~~maven~rules_android_maven'.
```

If I comment out line 66 in MODULE.bazel, I get the same error, except it's complaining about `build_bazel_rules_android` not visible instead. I am a little perplexed why the `@rules_android` repository resolution seems to work with bzlmod off, but not on.

I am on rules_jvm_external version 4.5, though I was getting the same failures on 6.0 (and I assume subsequent versions would also be affected).

**Comment by ted-xie** [comment]
cc @ahumesky

**Comment by shs96c** [comment]
This should already have been fixed in `rules_jvm_external` 6.1 by [this commit](https://github.com/bazelbuild/rules_jvm_external/commit/7785676c0ad7bfacc1e36023bfcaa4b98f217aeb). Closing this issue, but please comment if I am mistaken and I will reopen it.
